recipes/LibriSpeech/ASR/CTC/train_with_wav2vec_k2.py

```diff
@@ -0,0 +1,688 @@
+    )
+
+    return train_data, valid_data, test_datasets
+
+def get_lexicon(lang_dir, 
+                csv_files, 
+                extra_vocab_files, 
+                add_word_boundary=True):
+    '''
+    Read csv_files to generate a $lang_dir/lexicon.txt for k2 training.
+    This usually includes the csv files of the training set and the dev set in the output_folder.
+    During training, we need to make sure that the lexicon.txt contains all (or the majority of) 
+    the words in the training set and the dev set.
+
+    Args:
+    lang_dir: the directory to store the lexicon.txt
+    csv_files: a list of csv file paths 
+    extra_vocab_files: a list of extra vocab files, librispeech-vocab.txt is an example
+    add_word_boundary: whether to add word boundary symbols <eow> at the end of each line to the lexicon for every word
+
+    Note that in each csv_file, the first line is the header, and the remaining lines are in the following format:
+
+    ID, duration, wav, spk_id, wrd (transcription)
+
+    We only need the transcription in this function.
+
+    Returns:
+    None
+
+    Writes out $lang_dir/lexicon.txt
+
+    Note that the lexicon.txt is a text file with the following format:
+    word1 phone1 phone2 phone3 ...
+    word2 phone1 phone2 phone3 ...
+
+    In this code, we simply use the characters in the word as the phones.
+    You can use other phone sets, e.g., phonemes, BPEs, to train a better model.
+    '''
+    # Read train.csv, dev-clean.csv to generate a lexicon.txt for k2 training
+    lexicon = dict()
+    for file in csv_files:
+        with open(file) as f:
+            # Omit the first line
+            f.readline()
+            # Read the remaining lines
+            for line in f:
+                # Split the line 
+                _, _, _, _, trans = line.strip().split(",")
+                # Split the transcription into words
+                words = trans.split()
+                for word in words:
+                    if word not in lexicon:
+                        if add_word_boundary:
+                            lexicon[word] = list(word) + ["<eow>"]
+                        else:
+                            lexicon[word] = list(word)
+
+    for file in extra_vocab_files:
+        with open(file) as f:
+            for line in f:
+                # Split the line 
+                word = line.strip().split()[0]
+                # Split the transcription into words
+                if word not in lexicon:
+                    if add_word_boundary:
+                        lexicon[word] = list(word) + ["<eow>"]
+                    else:
+                        lexicon[word] = list(word)
+    # Write the lexicon to lang_dir/lexicon.txt
+    os.makedirs(lang_dir, exist_ok=True)
+    with open(os.path.join(lang_dir, "lexicon.txt"), "w") as f:
+        fc = "<UNK> <unk>\n"
+        for word in lexicon:
+            fc += word + " " + " ".join(lexicon[word]) + "\n"
+        f.write(fc)
+
+def arpa_to_fst(
```

Comment:
I don't think this could be moved to `librispeech_prepare` since it's very specific to the k2 version of the training script. On the other hand, it is very specific to librispeech so I don't think it can go inside speechbrain either.

Reply:
@TParcollet @mravanelli do you have an opinion on that please?

Reply:
Should we move it to `librispeech_prepare.py` or leave it here ?

Reply:
@geoph9 This should be moved to librispeech_prepare.py.

Reply:
@geoph9 it looks quite generic, are you sure it couldn't be part of the k2_integration folder?